parsec/core/sync_monitor.py

```diff
@@ -159,63 +158,66 @@ def set_remote_change(self, entry_id: EntryID) -> bool:
 
         now = timestamp()
 
-        if self.due_time > now:
+        # math.inf means due_time hasn't been set yet (or there is nothing
+        # to sync, in which case it's safe to continue the function)
+        if self.due_time != math.inf and self.due_time > now:
```

Comment:
Isn't it a bit counter intuitive? I don't understand why the `math.inf` special case is necessary, I would expect continuity over `due_time` here

Reply:
Yup I spend a good amount of time trying to make this clearer but didn't come to anything...

The idea is due_time can be:
- in the past, in which case we should do a synchronization
- in the future, in which case we have nothing to do yet
- in the infinite future, in which case we have nothing to do OR the due_time hasn't been computed

If due_time hasn't been computed, we should do the synchronization (given `self._refresh_checkpoint()` has just been called, `self.remote_changes` and `self.local_changes` has just been modified)
If we have nothing to do, we can also do a synchronization (which will do nothing and set the due_time to math.inf, so nothing is modified)

I considered using None as value for uninitialized due_time, but it feel really buggy (must check everywhere before doing comparisons)

Reply:
I think I'm missing the point of this extra logic... Why can't we stick to the existing logic of:
- Tick: is it time for me to wake up?
- If no: return
- If yes: synchronize what needs to be synchronize